geode-core/src/integrationTest/java/org/apache/geode/internal/cache/TXDetectReadConflictJUnitTest.java

```diff
@@ -102,4 +109,183 @@ public void testReadConflictsPR() throws Exception {
     assertEquals("value1", regionpr.get("key1"));
     mgr.commit();
   }
+
+
+  /**
+   * Test that two transactions with only read operations don't produce CommitConflictException
+   * This test fill some initial value on startup and after that create different threads with
+   * region#get operations
+   * Sync two different threads commit time through lock
+   */
+  @Test(/* no exception expected */)
+  public void testDetectReadConflict()
```

Comment:
Not sure if these unit tests actually would hit the locking timing during the commit.

I will check in a unit test that a transaction is committing while another transaction is holding the locks.
It will be good to add dunit tests (currently the tests are only run in unit test, it tests only local tx locks). 
The dunit tests could testing replicate regions and partition regions case (for testing distributed tx locks) and client server tests (testing ClientTXStateStub local locks.)